model/registry/cloudfoundry.yaml

```diff
@@ -0,0 +1,140 @@
+groups:
+  - id: registry.cloudfoundry
+    prefix: cloudfoundry
+    type: resource
+    brief: >
+      CloudFoundry resource attributes.
+    attributes:
+      - id: source.id
+        type: string
+        brief: >
+          A guid or another name describing the event source.
+        note: |
+          CloudFoundry defines the `source_id` in the [Loggegator v2 envelope](https://github.com/cloudfoundry/loggregator-api#v2-envelope).
+          It is used for logs and metrics emitted by CloudFoundry. It is
+          supposed to contain the application id for applications deployed on
+          the runtime or the component name, e.g. "gorouter", for CloudFoundry
+          components.
+
+          Application instrumentation should use the value from environment
+          variable `VCAP_APPLICATION.application_id`. This is the same value as
+          reported by `cf app <app-name> --guid`.
+
+          When system components are instrumented, values from the [Bosh spec](https://bosh.io/docs/jobs/#properties-spec)
+          should be used. The `source.id` should be set to
+          `spec.deployment/spec.name`.
+        examples:
+          - '218fc5a9-a5f1-4b54-aa05-46717d0ab26d'
+          - 'cf/gorouter'
+      - id: instance.id
```

Comment:
> An index or guid describing the concrete instance of the event source.

Is the `instance.id` referring to the actual app/resource? Or referring to something else? This part is a bit confusing, but maybe it's just my lack of CloudFoundry knowledge

> It is supposed to contain the application instance index for applications deployed on the runtime or the vm id for CloudFoundry components.

Reply:
`instance.id` refers to the VM that the app or system component is running on.

Reply:
This already is confusing in CloudFoundry: The instance.id can either be the app instance index starting from 0, if an app is deployed with multiple instances (similar to the index of a K8s StatefulSet). Or it can be the guid of a vm running a system component.

Picture the log stream of a typical CF application. It will contain logs of the application itself, where the instance id just contains the app instance index. But there will also be access logs from the Gorouter (ingress load balancer) which will contain the vm guid in the instance id. Consumers of CloudFoundry logs and metrics are used to this duality of use.